app/helpers/routing_helper.rb

```diff
@@ -14,4 +14,8 @@ def default_url_options
   def full_asset_url(source)
     Rails.configuration.x.use_s3 ? source : URI.join(root_url, ActionController::Base.helpers.asset_url(source)).to_s
   end
+
+  def hostname(url)
+    URI(url).host
```

Comment:
If you like `Addressable::URI.parse(url).display_uri.hostname` you can correspond to IDNA.